test: using globstar to ignore dist and node_modules

pudkrong opened the pull request:
This script is working on my macbook and it also works on CI (Ubuntu).

pudkrong committed:
test: using globstar to ignore dist and node_modules

coderbyheart reviewed (changes requested):
Please add a GitHub Actions workflow that demonstrates that unit tests work on OS X.

pudkrong committed:
test: run unit test on macos

- Remove duplicated code in another workflow
- Remove unsed environment variables

coderbyheart commented on .github/workflows/run-unit-test-on-macos.yaml:
```diff
@@ -0,0 +1,53 @@
+  issues: write
+  actions: write
+
+env:
+  CI: 1
+  FORCE_COLOR: 3
+  JSII_SILENCE_WARNING_UNTESTED_NODE_VERSION: 1
```
```suggestion
```

Not needed.

coderbyheart commented on .github/workflows/run-unit-test-on-macos.yaml:
```diff
@@ -0,0 +1,53 @@
+  test-on-macos:
+    runs-on: macos-latest
+
+    environment: ci
```
```suggestion
```

Not needed.

coderbyheart commented on .github/workflows/run-unit-test-on-macos.yaml:
```diff
@@ -0,0 +1,53 @@
+      - name: Install dependencies
+        run: npm ci --no-audit
+
+      - name: Compile TypeScript
+        run: npx tsc
+
+      - name: Check source code with eslint
+        run: npx eslint --fix --ext .js,.ts,.jsx,.tsx .
+
+      - name: Check if source code is properly formatted
+        run: npx prettier -c ./
+
```
```suggestion
```

Covered in other workflow.
